src/crypto/CHIPCryptoPALPSA.cpp

```diff
@@ -364,6 +364,21 @@ CHIP_ERROR HMAC_sha::HMAC_SHA256(const uint8_t * key, size_t key_length, const u
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR HMAC_sha::HMAC_SHA256(const Aes128KeyHandle & key, const uint8_t * message, size_t message_length, uint8_t * out_buffer,
+                                 size_t out_length)
+{
+    VerifyOrReturnError(isBufferNonEmpty(message, message_length), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(out_buffer != nullptr && out_length == PSA_HASH_LENGTH(PSA_ALG_SHA_256), CHIP_ERROR_INVALID_ARGUMENT);
+
+    const psa_algorithm_t algorithm = PSA_ALG_HMAC(PSA_ALG_SHA_256);
+    psa_status_t status             = PSA_SUCCESS;
+
+    status = psa_mac_compute(key.As<psa_key_id_t>(), algorithm, message, message_length, out_buffer, out_length, &out_length);
```

Comment:
If your plan is to pass a key handle created with `SessionKeystore::CreateKey` to this function, this will likely not work as `psa_mac_compute` verifies if the key has the type [HMAC](https://arm-software.github.io/psa-api/crypto/1.1/api/keys/types.html#c.PSA_KEY_TYPE_HMAC).

For the ICD use case, the code would need to add another handle type (for HMAC) and another `SessionKeystore::CreateKey` overload, and then the ICD core would need to maintain two key handles, for AES and HMAC, even though they would be created out of the same raw key.

Reply:
Duplicating the handle because of this limitation seems like a sub-optimal fix. That will bloat up the persistent storage needed for the ICD since it will add `sizeof( AESKeyHandle)` bytes per ICDMonitoringClient entry.

Isn't there another way without having to duplicate the Key handle ?

Reply:
change was done to support an HMAC key handle. The update to the ICDMonitoringTable will be done in a follow up PR.